lld/ELF/SyntheticSections.h

```diff
@@ -138,6 +137,14 @@ class GotSection final : public SyntheticSection {
     bool isSymbolFunc;
   };
   SmallVector<AuthEntryInfo, 0> authEntries;
+
+  // Map of GOT entries keyed by section, offset, and type. The purpose is to
```

Comment:
How do you propose targets override this to require additional properties be the same? For CHERI targets (currently downstream, but eventually to be upstreamed) we would also need to ensure the symbol size was the same, since that determines the bounds for the GOT entry.

Reply:
(along with additional complexity for handling "zero"-sized symbols to ensure that section start symbols get the bounds of the whole section, rather than be 0 bytes as they would otherwise be)

Reply:
I imagine in that case the second element of this tuple would evolve into something more complex. Right now it's just the offset. For CHERI targets, it would include size and other things necessary -- first and third element would still be the same.

Reply:
But how does that work when the type of the map needs to change? GotSection isn't templated on the target (and refactoring to do so would be a real pain I imagine).

Reply:
I think we would be okay if you add another member to the tuple named `size` (or just create a new struct) and populate that from `Symbol::getSize()` in `GotSection::addEntry()`. The original problem we are trying to solve (as mentioned in #129122) would still work.